ksql-engine/src/test/resources/query-validation-tests/key-field.json

```diff
@@ -481,13 +729,34 @@
       ],
       "post": {
         "issues": [
-          "key field of output has INCORRECT name: should be 'FOO'.",
-          "key field of output has INCORRECT type: should be 'INT'."],
+          "key field of output has INCORRECT name: should be 'FOO' - (see Issue #1 above)",
+          "key field of output has INCORRECT type: should be 'INT' - (see Issue #3 above)"
+        ],
         "sources": [
           {"name": "OUTPUT", "type": "table", "keyField": {"name": "KSQL_INTERNAL_COL_0", "schema": {"type": "STRING"}}}
         ]
       }
     },
+    {
+      "name": "table | initially null | group by (-) | key in value | no aliasing | legacy downstream query",
+      "statements": [
+        "CREATE TABLE INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key='foo', value_format='JSON');",
+        "CREATE TABLE INTERMEDIATE AS SELECT * FROM INPUT;",
```

Comment:
what value does `INTERMEDIATE` add above the tests outlined above?

Reply:
It's needed to create a source table with no key set. I've already documented this locally. If you don't mind, I'll add this in the next PR.